Tests/Realm.Tests/Database/RealmValueTests.cs

```diff
@@ -402,9 +411,9 @@ private void RunNumericTests(RealmValue rv, long value, bool isManaged)
 
             Assert.That(rv.Type, Is.EqualTo(RealmValueType.Object));
 
-            Assert.That((RealmObjectBase)rv, Is.EqualTo(value));
-            Assert.That(rv.As<RealmObjectBase>(), Is.EqualTo(value));
-            Assert.That(rv.AsRealmObject(), Is.EqualTo(value));
+            Assert.That(rv.AsIRealmObject(), Is.EqualTo(value));
```

Comment:
It should, but unfortunately we can't do castings anymore, because we can't have implicit/explicit conversion to/from interfaces, so it can be done. I'll remove this line, given somethign similar is happening some line down